### PR TITLE
receiving of groups refactored

### LINKED ISSUE #239: Refactor receiving of groups
Make getting of groups only sorted by order

### PR DESCRIPTION
**Pull request by petrokrechunyak**

### CODE REVIEW

File: frontend/.eslintrc.json [hunk 1]
```diff
@@ -54,7 +54,7 @@
       }
     ]
   },
-  "parser": "babel-eslint",
+  "parser": "@babel/eslint-parser",
```
Comment (dshma) [inline frontend/.eslintrc.json]: I don't think it has anything to do with this task, do create a separate one to update obsolete dependencies if needed
FYI:
https://github.com/nromanen/Schedule/issues/241

File: frontend/package.json [hunk 2]
```diff
@@ -5,32 +5,32 @@
     "dependencies": {
         "@babel/core": "^7.16.0",
         "@date-io/moment": "1.3.13",
-        "@kenshooui/react-multi-select": "1.1.6",
+        "@kenshooui/react-multi-select": "^1.0.53",
```
Comment (dshma) [inline frontend/package.json]: the same comment applies to the changes within this class

File: frontend/src/components/Auth/Auth.js [hunk 3]
```diff
@@ -163,12 +163,12 @@ const Auth = (props) => {
             document.title = t(LOGIN_TITLE);
             return (
                 <div className="auth-container">
-                    <a
-                        className="hidden-link"
-                        href={`${process.env.REACT_APP_API_BASE_URL.trim()}${`/${GOOGLE_LOGIN_URL}`}`}
-                    >
-                        auth via google
-                    </a>
+                    {/*<a*/}
```
Comment (dshma) [inline frontend/src/components/Auth/Auth.js]: why comment this?

File: frontend/src/constants/axios.js [hunk 4]
```diff
@@ -36,8 +36,7 @@ export const SCHEDULE_CHECK_AVAILABILITY_URL = 'schedules/data-before';
 export const BUSY_ROOMS = 'schedules/full/rooms';
 export const ROOMS_AVAILABILITY = 'rooms/available';
 export const SEMESTERS_URL = 'semesters';
-export const GROUPS_URL = 'groups';
-export const GROUPS_ORDERED_URL = 'groups/ordered';
+export const GROUPS_URL = 'groups'
```
Comment (dshma) [inline frontend/src/constants/axios.js]: missing semicolon

File: src/main/java/com/softserve/controller/GroupController.java [hunk 5]
```diff
@@ -35,7 +35,9 @@ public GroupController(GroupService groupService, GroupMapper groupMapper) {
     @ApiOperation(value = "Get the list of all groups")
     public ResponseEntity<List<GroupDTO>> getAll() {
         log.info("In getAll ()");
-        return ResponseEntity.status(HttpStatus.OK).body(groupMapper.groupsToGroupDTOs(groupService.getAll()));
+        return ResponseEntity
+                .status(HttpStatus.OK)
+                .body(groupMapper.groupsToGroupDTOs(groupService.getAllBySortingOrder()));
```
Comment (dshma) [inline src/main/java/com/softserve/controller/GroupController.java:40]: I don't think it's a good idea to force sorting by order in any case, it should rather be a default sort if none specified

File: src/main/resources/hibernate.properties [hunk 6]
```diff
@@ -2,8 +2,8 @@ hibernate.dialect=org.hibernate.dialect.PostgreSQL9Dialect
 
 hibernate.connection.driver_class=org.postgresql.Driver
 hibernate.connection.url=jdbc:postgresql://localhost:5432/schedule
-hibernate.connection.username=postgres
-hibernate.connection.password=10772102
+hibernate.connection.username=ama_user
```
Comment (dshma) [inline src/main/resources/hibernate.properties]: no way this is getting committed, this is your local server's config

File: src/main/resources/mail.properties [hunk 7]
```diff
@@ -1,4 +1,4 @@
-spring.mail.host=smtp.office365.com
+spring.mail.host=smtp.gmail.com
```
Comment (dshma) [inline src/main/resources/mail.properties:1]: why change this?

**Review by dshma — CHANGES_REQUESTED** [review]

File: src/main/resources/mail.properties (hunk 7 above)
Comment (petrokrechunyak) [reply src/main/resources/mail.properties:1]: We use our own mail accounts as mail sender to students

File: src/main/java/com/softserve/controller/GroupController.java (hunk 5 above)
Comment (petrokrechunyak) [reply src/main/java/com/softserve/controller/GroupController.java:40]: My task was to change default getter of groups to sorted

File: frontend/src/components/Auth/Auth.js (hunk 3 above)
Comment (petrokrechunyak) [reply frontend/src/components/Auth/Auth.js]: Last group started to do Google authentication but didnt end, and now without commenting this link we cant go to login page

**Review by dshma — APPROVED** [review]

**Pull request merged**